OpenTrons/opentrons-api
# fix(app): never show module calibration banner for OT-2-specific modules

closes [RQA-1527](https://opentrons.atlassian.net/browse/RQA-1527)

# Overview

On ModuleCard, a calibration banner is set to render if there is no last modified date for module offsets. Here I add a check to only render the calibration banner if the module is not a member of OT-2-specific modules (those that will fail protocol analysis on a Flex and cannot be calibrated):
- magnetic module v1
- magnetic module v2
- temperature module v1
- thermocycler module v1

# Test Plan

- plug in one of the above OT-2-specific modules to a Flex
- navigate to the Flex device details page
- verify that the module card does not render a calibration banner
<img width="998" alt="Screen Shot 2024-01-08 at 12 17 22 PM" src="https://github.com/Opentrons/opentrons/assets/47604184/a23556bd-a89f-4fcc-8fff-dda17ee04d86">

# Changelog

- create constant for OT-2-specific modules
- add robot + module check for rendering calibration banner on module card
- add test for OT-2-specific module plugged into a Flex

# Risk assessment

low

[RQA-1527]: https://opentrons.atlassian.net/browse/RQA-1527?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
